cpu/sam0_common/periph/eth.c

```diff
@@ -0,0 +1,310 @@
+/*
+ * Copyright (C) 2020 Mesotic SAS
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup     cpu_sam0_common
+ * @{
+ *
+ * @file
+ * @brief       Low-level Ethernet driver implementation
+ *
+ * @author      Dylan Laduranty <dylan.laduranty@mesotic>
+ *
+ * @}
+ */
+
+#include "luid.h"
+
+#include "iolist.h"
+#include "net/gnrc/netif/ethernet.h"
+#include "net/gnrc.h"
```

Comment:
do you really need those?
I don't see you calling any `gnrc_` functions here - and this driver should also work with LWIP